recipes/pycraf/meta.yaml

```diff
@@ -0,0 +1,73 @@
+{% set name = "pycraf" %}
+{% set version = "0.25.7" %}
+{% set sha256 = "f17bf89af7f9a708a6aa1b79ad24a48a09754dbe39a1edf03b27f78d7bb70820" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [py27]
+  # script: "{{ PYTHON }} -m pip install . --no-deps -vv"
```

Comment:
Why are you not using pip?

Reply:
First, let me thank you a lot for the review, @ocefpaf. I appreciate your help very much. It's the first time, I do a conda recipe, please be patient with me.

With `script: "{{ PYTHON }} -m pip install . --no-deps -vv"` I get an error message:
```
RuntimeError: Setuptools downloading is disabled in conda build. Be sure to add all dependencies in the meta.yaml  url=https://pypi.python.org/simple/astropy-helpers/
```
On my local system, I get the same error when I run `conda build`. What works is `script: "{{ PYTHON }} setup.py install --record record.txt --offline"` (with the two parameters being essential).

I think, the problem is that `astropy-helpers` is not available. I tried to add it to the requirements, but it seems not to be available on conda-forge. Even then, I'm not sure it would work that way. Pycraf is using the [Astropy affiliated package template](https://github.com/astropy/package-template). `astropy-helpers` is a git submodule and fetched during the installation process (I think). I checked various other recipes that are also based on the package template, all of them use the `script: "{{ PYTHON }} setup.py install --record record.txt --offline"` approach. Perhaps @bsipocz can comment on this?

Reply:
What about other packages that use the same template? For example [astropy-healpix](/conda-forge/astropy-healpix-feedstock/tree/master/recipe) also uses `python setup.py install` in the recipe...

Reply:
If there is a download during the build time it should be fixed too.

Reply:
Not sure if that is the case at all. The `astropy-helper` submodule is shipped as part of the `pycraf.tar.gz`. I'll investigate this in more detail, when I'm back at my computer.

Reply:
No, there is no download involved. `astropy-helpers` is a submodule, but is bundled into the sdist. The infrastructure is set up though as it tries to download the submodule, unless ``--offline`` is specified. 

We have no interest of getting astropy-helpers packaged up, doing so actually is expected to cause more troubles as it would solve.

Reply:
In that case there is probably the same invocation for `pip`. Note that build packages with `python setup.py` is deprecated and `pip` is the way to go as PyPA changes to new formats. With that said, we can open exceptions until the package is fixed upstream and/or we figure out the same `pip` invocation for that.

Reply:
Well, I tried several potential solutions, without success so far. It appeared to me that the `--install-option` parameter could be used to hand over the necessary `setup.py` options, e.g., 
```
script: '{{ PYTHON }} -m pip install --install-option="--single-version-externally-managed" --install-option="--record record.txt" --install-option="--offline" .'
```
but I still get the same error (from the log, it seems that only `python setup.py egg_info` is invoked when the error is thrown, but none of the above options appears.

According to (https://github.com/pypa/pip/issues/1707) it's a little bit more complicated (essentially, they propose to use `--global-option` instead of `--install-option` and add `--global-option "build_ext"), but I couldn't make this work either.

So, unless you got another idea, perhaps we can really use the `python setup.py` approach, even though it is deprecated.

Reply:
> So, unless you got another idea, perhaps we can really use the python setup.py approach, even though it is deprecated.

That is what I said above :smile: sorry if it was not clear.

Reply:
We have tried to get this done using pip, but haven't yet found a way where it works. So, please ping Matt or I if you anywhere run into a command that works 😄 

On the other hand, there will be probably efforts put in the direction of removing this submodule dependency in astropy land, that would also provide a natural solution for this problem.